src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -289,9 +289,20 @@ private int AppendInnerExceptionTree(Exception currentException, int currentLeve
 
         private void AppendInnerException(Exception currentException, StringBuilder builder)
         {
-            // separate inner exceptions
+            var formats = InnerFormats;
+            if (formats == null)
+            {
+                //Don't flow toString to inner exceptions if used for out exception
+                formats = new List<ExceptionRenderingFormat>(Formats);
```

Comment:
Think it is sad that you allocate a new list for every LogEvent. Could it not be created once, when parsing the Format?

Reply:
Can do. Wasn't sure if the approach now is good enough

Reply:
Ok, then I will wait until it is good enough :)

Reply:
Do you think this approach is good enough?

Reply:
I was thinking that if were going to dive into the Exception-object and display all inner-exceptions with details, then it would be nice it would be displayed nicely. Like here https://github.com/oleksabor/nlog.indentexception

Since the layout capability of the Format-specifier is rather limited, then it I think works best with a single exception-object. But besides `Format=ToString`, then you probably should also handle `Format=@` (Json)

Reply:
But actually I think it is a bad approach to modify Format-specifier for inner exceptions. Instead MaxInnerExceptionLevel should be nullable. And when not configured explicitly then it should default to 10 (But if Format contains ToString or @ then it should default to 0)

Reply:
Another example where prettify exception is better than Exception.ToString: https://codereview.stackexchange.com/questions/147768/exceptionprettifier-for-prettier-exception-strings

Reply:
> Another example where prettify exception is better than Exception.ToString: https://codereview.stackexchange.com/questions/147768/exceptionprettifier-for-prettier-exception-strings

Agree on that one, but sounds like a scope creep to me

Reply:
Yes. But trying to argue for that changing `MaxInnerExceptionLevel` gives more side-effects than wins. Maybe making this improvement by doing something new/different.

Reply:
I agree this is difficult. I think ${exception} has at least 2 use cases, and that's impossible to handle in one renderer.

So maybe we need to split it to separate renderers, like ${fancy-exception} and ${all-exceptions).

Any way the issue here I try to solve is that people are lossing their inner exceptions. That's now partly covered with the ToString, but not for the exception data

Reply:
There is a "gateway" in the `Format`-property for `${exception}`, which can be (ab)used like I did when adding `format=@` (automatically renders all inner-exceptions even when not requested).